pkg/clients/aws/route53.go

```
@@ -337,3 +485,44 @@ func NewClient(kubeClient client.Client, secretName, namespace, region string) (
 
 	return c, err
 }
+
+func getSTSCredentials(client *sts.STS, roleArn string, externalID string, roleSessionName string) (*sts.AssumeRoleOutput, error) {
+	// Default duration in seconds of the session token 3600. We need to have the roles policy
+	// changed if we want it to be longer than 3600 seconds
+	var roleSessionDuration int64 = 3600
+	fmt.Printf("Creating STS credentials for AWS ARN: %s", roleArn)
+	// Build input for AssumeRole
+	assumeRoleInput := sts.AssumeRoleInput{
+		DurationSeconds: &roleSessionDuration,
+		RoleArn:         &roleArn,
+		RoleSessionName: &roleSessionName,
+	}
+	if externalID != "" {
+		assumeRoleInput.ExternalId = &externalID
+	}
+
+	assumeRoleOutput := &sts.AssumeRoleOutput{}
+	var err error
+	for i := 0; i < 100; i++ {
+		time.Sleep(500 * time.Millisecond)
+		assumeRoleOutput, err = client.AssumeRole(&assumeRoleInput)
+		if err == nil {
+			break
+		}
+		if i == 99 {
+			fmt.Printf("Timed out while assuming role %s", roleArn)
```

We should probably return an error here.

cc @iamkirkbater for AAO too.